File: client/src/main/java/au/com/codeka/warworlds/game/starfield/scene/TacticalTextureLruCache.java
package au.com.codeka.warworlds.game.starfield.scene;

import androidx.annotation.NonNull;
import androidx.collection.LruCache;

import au.com.codeka.common.Pair;

class TacticalTextureLruCache extends LruCache<Pair<Long, Long>, TacticalTexture> {
  TacticalTextureLruCache(int maxSize) {
    super(maxSize);
  }

  @Override
  public void entryRemoved(
      boolean evicted, @NonNull Pair<Long, Long> key, TacticalTexture oldValue,
      TacticalTexture newValue) {
    oldValue.close();
  }
}

File: client/src/main/kotlin/au/com/codeka/warworlds/client/game/starfield/scene/TacticalTextureLruCache.kt
package au.com.codeka.warworlds.client.game.starfield.scene

import androidx.collection.LruCache
import au.com.codeka.warworlds.common.proto.SectorCoord

class TacticalTextureLruCache(maxSize: Int) : LruCache<SectorCoord, TacticalTexture>(maxSize) {
  override fun entryRemoved (
      evicted: Boolean, key: SectorCoord, oldValue: TacticalTexture, newValue: TacticalTexture?) {
    oldValue.close()
  }
}